helpers.js

```diff
@@ -0,0 +1,33 @@
+const fs = require("fs");
+const { access } = require("fs/promises");
+
+async function assertPathsExistence(paths) {
+  await Promise.all(
+    paths.map(async (path) => {
+      try {
+        await access(path, fs.constants.F_OK);
+        return true;
+      } catch {
+        throw new Error(`Path ${path} does not exist on the agent`);
+      }
+    }),
+  );
+}
+
+function sanitizeHostname(hostname) {
+  let sanitizedHostname = hostname;
+
+  if (!/^https?:\/\//i.test(hostname)) {
```

Comment:
Plugin needs to enforce using `https`, otherwise Testrail API is not working correctly. That's why we have two checks here:
* Checking if `hostname` starts with `http`, if yes then change it to `https`
* Checking if `hostname` starts with `http` or `https`, if not then prepend it with `https`, without the protocol API requests don't even get sent

Reply:
First check executes regexp only for the sake of merging two conditions into one. Regexps are harder to read than regular code and could be performance heavy. The same could be achieved by two `!sanitizedHostname.startsWith` calls.
Then we could see that second condition will never trigger when first does.

This would be much more transparent:
`
let sanitizedHostname = null;

if (hostname.startsWith("https://")) {
    sanitizedHostname = hostname;
} else if (hostname.startsWith("http://")) {
    sanitizedHostname = hostname.replace("http://","https://");
} else {
    sanitizedHostname = `https://${hostname}`;
}

return sanitizedHostname;
`